Framework/Algorithms/src/AbsorptionCorrectionPaalmanPings.cpp

```diff
@@ -0,0 +1,511 @@
+                    sample_container_L2s, 0, m_numSampleVolumeElements);
+      assY[j] =
+          integral / m_sampleVolume; // Divide by total volume of the shape
+      asscY[j] =
+          crossIntegral / m_sampleVolume; // Divide by total volume of the shape
+
+      integral = 0.0;
+      crossIntegral = 0.0;
+      doIntegration(integral, crossIntegral, -containerLinearCoefAbs[j],
+                    m_containerLinearCoefTotScatt, m_containerElementVolumes,
+                    m_containerL1s, container_L2s, -sampleLinearCoefAbs[j],
+                    m_ampleLinearCoefTotScatt, m_container_sampleL1s,
+                    container_sample_L2s, 0, m_numContainerVolumeElements);
+      accY[j] =
+          integral / m_containerVolume; // Divide by total volume of the shape
+      acscY[j] = crossIntegral /
+                 m_containerVolume; // Divide by total volume of the shape
+
+      // Make certain that last point is calculated
+      if (m_xStep > 1 && j + m_xStep >= specSize && j + 1 != specSize) {
+        j = specSize - m_xStep - 1;
+      }
+    }
+
+    // Interpolate linearly between points separated by m_xStep,
+    // last point required
+    if (m_xStep > 1) {
+      auto histnew = ass->histogram(i);
+      interpolateLinearInplace(histnew, m_xStep);
+      ass->setHistogram(i, histnew);
+      auto histnew2 = assc->histogram(i);
+      interpolateLinearInplace(histnew2, m_xStep);
+      assc->setHistogram(i, histnew2);
+      auto histnew3 = acc->histogram(i);
+      interpolateLinearInplace(histnew3, m_xStep);
+      acc->setHistogram(i, histnew3);
+      auto histnew4 = acsc->histogram(i);
+      interpolateLinearInplace(histnew4, m_xStep);
+      acsc->setHistogram(i, histnew4);
+    }
+
+    prog.report();
+
+    PARALLEL_END_INTERUPT_REGION
+  }
+  PARALLEL_CHECK_INTERUPT_REGION
+
+  g_log.information() << "Total number of elements in the integration was "
+                      << m_sampleL1s.size() << '\n';
+
+  const std::string outWSName = getProperty("OutputWorkspace");
+
+  std::vector<std::string> names;
+  names.emplace_back(outWSName + "_ass");
+  API::AnalysisDataService::Instance().addOrReplace(names.back(), ass);
+  names.emplace_back(outWSName + "_assc");
+  API::AnalysisDataService::Instance().addOrReplace(names.back(), assc);
+  names.emplace_back(outWSName + "_acc");
+  API::AnalysisDataService::Instance().addOrReplace(names.back(), acc);
+  names.emplace_back(outWSName + "_acsc");
+  API::AnalysisDataService::Instance().addOrReplace(names.back(), acsc);
+
+  auto group = createChildAlgorithm("GroupWorkspaces");
+  group->initialize();
+  group->setProperty("InputWorkspaces", names);
+  group->setProperty("OutputWorkspace", outWSName);
+  group->execute();
+  API::WorkspaceGroup_sptr outWS = group->getProperty("OutputWorkspace");
+
+  setProperty("OutputWorkspace", outWS);
+
+  // Now do some cleaning-up since destructor may not be called immediately
```

Comment:
I'm suspicious this comment is incorrect. C++ is quite deterministric.

@williamfgc thoughts?

Reply:
This was just copied over from https://github.com/mantidproject/mantid/blob/67b7ba4c90a4701cbe628974409dbff6253e7dc9/Framework/Algorithms/src/AbsorptionCorrection.cpp#L297 so I don't know.

Reply:
FYI, [clear]([(http://www.cplusplus.com/reference/vector/vector/clear/) doesn't necessarily reclaim the memory (deallocate). The only thing that is guarantee is the logic of having an empty container.